tests/alternative/Network.lua

```diff
@@ -1,155 +1,352 @@
+
+			proj.file:delete()
+			File(neterrorLayer.file):delete()
+
+			local disconnectedError = function()
+				local roads = CellularSpace{
+					file = roadsCurrDir,
+					missing = 0
+				}
+
+				Network{
+					lines = roads,
+					target = communities,
+					progress = false,
+					inside = function(distance) return distance end,
+					outside = function(distance) return distance * 2 end,
+					error = 900
+				}
+			end
+
+			unitTest:assertError(disconnectedError, "The network is disconnected. Data 'neterror.shp' was automatically created with attribute 'net_id' containing the separated networks.")
+
+			local neterror = CellularSpace{
+				file = "neterror.shp",
+				missing = 0
+			}
+
+			unitTest:assertNotNil(neterror:sample().net_id)
+
+			File(roadsCurrDir):delete()
+			neterror.file:delete()
 		end
 
-		unitTest:assertError(error_func, "Line: '7' does not touch any other line. The minimum distance found was: 843.46359196883.")
+		disconnectedNetworkTest()
+
+		local errorArgumentError = function()
+			local roadsSouth = CellularSpace{
+				file = filePath("test/roads_sirgas2000_south3.shp", "gpm")
+			}
+
+			local ports = CellularSpace{
+				file = filePath("test/porto_alegre_sirgas2000.shp", "gpm"),
+				missing = 0
+			}
 
-		error_func = function()
 			Network{
-				lines = roads,
-				target = communities,
-				inside = function(distance) return distance end,
-				outside = function(distance) return distance * 2 end,
-				error = 900
+				lines = roadsSouth,
+				target = ports,
+				progress = false,
+				error = 400,
+				inside = function(distance)
+					return distance
+				end,
+				outside = function(distance)
+					return distance * 4
+				end
 			}
 		end
 
-		unitTest:assertError(error_func, "The network is disconnected. For example, objects '1' and '2' belong to two separated networks.")
+		unitTest:assertError(errorArgumentError, "Line '47' was added because the value of argument 'error: 400'. Remove the error argument and correct the lines disconnected.")
 
-		roads = CellularSpace{
-			file = filePath("error/".."roads_overlay_points.shp", "gpm"),
-			missing = 0
-		}
+		local lineCrossesError = function()
+			local roads = CellularSpace{
+				file = filePath("test/roads_sirgas2000_south8.shp", "gpm")
+			}
+
+			local ports = CellularSpace{
+				file = filePath("test/port_antonina_sirgas2000.shp", "gpm"),
+				missing = 0
+			}
 
-		error_func = function()
 			Network{
 				lines = roads,
-				target = communities,
-				inside = function(distance) return distance end,
-				outside = function(distance) return distance * 2 end
+				target = ports,
+				progress = false,
+				inside = function(distance)
+					return distance
+				end,
+				outside = function(distance)
+					return distance * 4
+				end
 			}
 		end
 
-		unitTest:assertError(error_func, "Lines '6' and '14' cross each other.")
+		unitTest:assertError(lineCrossesError, "Line '26' crosses lines '20' and '18' in their endpoints. Please, split line '26'.")
```

Comment:
Split in two?

Reply:
"Line '26' crosses touching lines '20' and '18' in their endpoints. Please, split line '26' in two where they cross."